### PR TITLE
Firefox fix for ImageRecorder (+small refactor)

### PR DESCRIPTION
**Pull request by maartenbreddels**

### CODE REVIEW

File: js/src/webrtc.js [hunk 1]
```diff
@@ -605,10 +599,8 @@ var ImageRecorderModel = RecorderModel.extend({
             // turn the mediastream into a video element
             let video = document.createElement('video');
             video.srcObject = mediaStream;
+            video.play()
             return utils.onCanPlay(video).then(() => {
```
Comment (martinRenou) [inline js/src/webrtc.js]: maybe you can use await here as well

File: js/src/webrtc.js [hunk 2]
```diff
@@ -647,7 +636,8 @@ var ImageRecorderModel = RecorderModel.extend({
         if (this.get('_data_src') != '') {
             URL.revokeObjectURL(this.get('_data_src'));
         }
-        this.snapshot()
+        if(this.get('recording'))
+            this.snapshot()
```
Comment (martinRenou) [inline js/src/webrtc.js]: Nice! I forgot to tell you but I saw this one as well yesterday :)

**Comment by maartenbreddels** [comment]
guess we can do more of those, maybe in a separate pr?

**Comment by martinRenou** [comment]
I agree :) If we can get rid of each `then` call, the code will be much more readable

**Pull request merged**